src/test/java/net/minecraftforge/test/EnumHelperTest.java

```diff
@@ -16,15 +17,68 @@
     private boolean failed = false;
 
     @Test
-    public void test()
+    public void testEnumHelperMethodsMatchEnumConstructors()
+    {
+        //Quick note to point out that Enum constructors start with (string name, int ordinal).
+        //  EnumHelper methods by convention start with (string name), and the ordinal is calculated.
+        for (Method method : EnumHelper.class.getDeclaredMethods())
+        {
+            String name = method.getName();
+            if (!name.equals("addEnum") && name.startsWith("add"))
+            {
+                System.out.println("  " + method);
+
+                Class<?> returnType = method.getReturnType();
+                Constructor<?>[] declaredConstructors = returnType.getDeclaredConstructors();
+                Class<?>[] actualParameters = method.getParameterTypes();
+
+                Assert.assertTrue(actualParameters.length >= 1);
+                Assert.assertEquals("First parameter of add method should be String (name)",
+                        String.class,
+                        actualParameters[0]);
+                Assert.assertEquals(String.class, actualParameters[0]);
+
+                for (Constructor<?> declaredConstructor : declaredConstructors)
+                {
+                    System.out.println("    " + declaredConstructor.toString());
+
+                    Class<?>[] expectedParameters = declaredConstructor.getParameterTypes();
+
+                    Assert.assertTrue(expectedParameters.length >= 2);
+                    Assert.assertEquals("First parameter of enum constructor should be String (name)",
+                            String.class,
+                            expectedParameters[0]);
+                    Assert.assertEquals("Second parameter of enum should be int (ordinal)",
+                            int.class,
```

Comment:
Indeed, Enums are simply syntax sugar.
They are short hand for 'public static final TYPE NAME = new TYPE("NAME", id++);'
This is how we are able to reverse all the enum values to their true names in MCP, because they are named in code as a string, and that doesn't get obfuscated.
If you'll notice in the EnumHelper it prepends the name and new ordinal number.